src/prototype_room_external.js

```diff
@@ -296,24 +294,31 @@ Room.prototype.handleReservedRoom = function() {
         }
       };
       // TODO move the creep check from the reserver to here and spawn only sourcer (or one part reserver) when controller.level < 4
-      let energyThreshold = 1300;
 
       // TODO Not sure when this happens, Check the closest room and set reservation
       if (!this.memory.reservation) {
         this.log('No reservation');
-        return;
+        return false;
       }
 
-      if (Game.rooms[this.memory.reservation.base].misplacedSpawn) {
-        energyThreshold = 1600;
+      let reservation = this.memory.reservation;
+      let energyNeeded = 1300;
```

Comment:
Maybe this can be extracted to a `config` value and `https://github.com/TooAngel/screeps/blob/master/src/prototype_room_utils.js#L3` to check the misplacedSpawn thingy,